File: server/game/lib/achievement.cpp
#include <fb/game/achievement.h>

using namespace fb::game;

achievement::achievement(uint32_t id, std::string text, uint8_t icon, uint16_t color)
    : id(id)
    , text(std::move(text))
    , icon(icon)
    , color(color)
{
}

File: server/http/Model/Achievement.cs
using StackExchange.Redis;

namespace Http.Model
{
    public class AchievementKey : BaseModel, IRedisHashKey
    {
        public required uint Uid { get; set; }
        public uint Id { get; set; }
        public uint? GetHash() => Uid;

        public RedisKey GetRedisKey() => $"cache:achievement:{Uid}";

        public RedisValue GetRedisField() => $"{Id}";
    }

    public class Achievement : AchievementKey, IModel
    {
        public string Text { get; set; } = string.Empty;
        public byte Icon { get; set; }
        public ushort Color { get; set; }
    }
}
